lib/api/actions/functions.php

```diff
@@ -410,119 +410,189 @@ function beans_reset_action( $id ) {
 }
 
 /**
- * Get action.
+ * Get the action's configuration for the given ID and status. Returns `false` if the action is not registered with
+ * Beans.
  *
+ * @since  1.0.0
+ * @since  1.5.0 Made WPCS compliant.
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param string $status Status for which to get the action.
+ *
+ * @return array|bool
  */
 function _beans_get_action( $id, $status ) {
-
 	global $_beans_registered_actions;
```

Comment:
Since we are making improvements here, I think it would be good to get rid of the global and use WP Cache API instead. Example:

``` PHP
wp_cache_set( '_registered_actions', $data, 'beans' );
```
``` PHP
wp_cache_get( '_registered_actions', 'beans' );
```

By doing this, we can get rid of the json encoding which was originally used for performance purposes. This will allow users to take advance of anonymous callback such as the example below which is currently not possible.

``` PHP
beans_modify_action( 'beans_post_image', null, function() {
	echo 'test';
} );
```

Since the `$_beans_registered_actions` was private, we should need to bother about backwards compat.

Reply:
I was thinking about rolling these types of changes into Beans 2, as we look at the private functionality and global variables.  

>This will allow users to take advance of anonymous callback such as the example below which is currently not possible.
This is not possible with PHP 5.2.  But when we leap forward to PHP 5.6, then this shift makes a lot more sense.

Reply:
Sure, that is ok.